### PR TITLE
Use Docker for Mac's NFS support

### PR DESCRIPTION
**Pull request by acbramley**
Based on https://medium.com/@sean.handley/how-to-set-up-docker-for-mac-with-native-nfs-145151458adc

We get huge benefits from this:

Unit test current baseline: 25 seconds (in web container)
Robo build baseline: 7 minutes 25 seconds

Unit test with NFS: 6.3 seconds (400% speed up)
Robo build with NFS: 3 minutes 5 seconds (>200% speed up)

The `./dsh nfs` script is a one off script to run to set up NFSD configuration.

### CODE REVIEW

**Comment by singularo** [comment]
Shouldn't we be continuing to move setup style tasks out of dsh?
Perhaps calling seperate setup scripts for various scenarios, like nfs on mac, nginx on linux, etc..?

**Comment by acbramley** [comment]
@singularo I'd be happy to leave it out as well, not sure where else it'd live though.

**Comment by pingers** [comment]
All I can think is we do some "yucky" include based on "if osx then include dsh.osx". Hmm

@fenetikm @bart88 mind trying this some time?

File: dsh [hunk 1]
```diff
@@ -108,6 +108,78 @@ dsh_pull() {
...
+dsh_setup_nfs() {
+  if [ ${HOST_TYPE} != "mac" ]; then
+    echo "This script is OSX-only. Please do not run it on any other Unix."
+    exit 1
+  fi
+
+  if [[ $EUID -eq 0 ]]; then
+    echo "This script must NOT be run with sudo/root. Please re-run without sudo." 1>&2
+    exit 1
+  fi
```
Comment (pingers) [inline dsh]: Do we also need a check for NFS being installed? I have no idea if it's in OSX by default.

Comment (acbramley) [reply dsh]: Pretty sure it's shipped by default

**Comment by acbramley** [comment]
Other thing we could do is link to some documentation that contained the script.

File: dsh (hunk 1 above)
Comment (fenetikm) [reply dsh]: yep 'tis

**Comment by fenetikm** [comment]
The only issue I am getting with this is that if one wants to then use something like Vagrant (e.g. legacy existing app) one has to clean up the exports file - maybe we could have an opposite command in there to remove itself?

**Comment by pingers** [comment]
@acbramley how do you feel about @fenetikm's idea? Something you can do?

**Comment by acbramley** [comment]
@pingers I can look into it for sure

**Comment by acbramley** [comment]
@fenetikm @pingers New command `./dsh rnfs` added to revert the changes to the exports file.

**Comment by pingers** [comment]
Nice one! @fenetikm mind taking a look seeing I can't test.

**Comment by pingers** [comment]
:shipit: sorry about the delay. No one got to it, but I think it's in a good place. Let's go with it. It's a dev tool, we can always make more changes if needed.

**Pull request merged**